File: mc/rmcp/src/main/java/calebxzhou/rdi/mc/common2/mcp/RMcpEndpointException.java
package calebxzhou.rdi.mc.common2.mcp;

public class RMcpEndpointException extends RuntimeException {
    private final String code;

    public RMcpEndpointException(String code) {
        super(code);
        this.code = code;
    }

    public RMcpEndpointException(RErrorCode code) {
        this(code.id());
    }

    public String code() {
        return code;
    }
}

File: mc/rmcp/common/src/main/kotlin/calebxzhou/rdi/mc/common2/mcp/RMcpEndpointException.kt
package calebxzhou.rdi.mc.common2.mcp

open class RMcpEndpointException(val code: String?, val detail: String) : RuntimeException(code) {
    constructor(code: RErrorCode) : this(code.id,code.info)
    constructor(ex: RMcpEndpointException) : this(ex.javaClass.simpleName,ex.detail)
    constructor(detail: String) : this(null,detail)

    fun code(): String {
        return code?:""
    }
}
